plugins/inventory/opennebula.py

```diff
@@ -0,0 +1,225 @@
+
+# Pass a label filter to the API
+plugin: opennebula
+api_url: https://opennebula:2633/RPC2
+filter_by_label: Cache
+'''
+
+try:
+    import pyone
+
+    HAS_PYONE = True
+except ImportError:
+    HAS_PYONE = False
+
+from ansible.errors import AnsibleError
+from ansible.plugins.inventory import BaseInventoryPlugin, Constructable
+from ansible.module_utils._text import to_native
+
+import os
+
+
+def _get_connection_info(url, username, password, authfile):
+    if not username:
+        if not password:
+            if authfile is None:
+                authfile = os.path.join(os.environ.get("HOME"), ".one", "one_auth")
+            try:
+                with open(authfile, "r") as fp:
+                    authstring = fp.read().rstrip()
+                username = authstring.split(":")[0]
+                password = authstring.split(":")[1]
+            except (OSError, IOError):
+                raise AnsibleError("Could not find or read ONE_AUTH file at '{e}'".format(e=authfile))
+            except Exception:
+                raise AnsibleError("Error occurs when read ONE_AUTH file at '{e}'".format(e=authfile))
+
+    from collections import namedtuple
+
+    auth_params = namedtuple('auth', ('url', 'username', 'password'))
+
+    return auth_params(url=url, username=username, password=password)
+
+
+def _get_vm_ipv4(vm):
+    nic = vm.TEMPLATE.get('NIC')
+
+    if isinstance(nic, dict):
+        nic = [nic]
+
+    for net in nic:
+        return net['IP']
+
+    return False
+
+
+def _get_vm_ipv6(vm):
+    nic = vm.TEMPLATE.get('NIC')
+
+    if isinstance(nic, dict):
+        nic = [nic]
+
+    for net in nic:
+        if net.get('IP6_GLOBAL'):
+            return net['IP6_GLOBAL']
+
+    return False
+
+
+def _retrieve_servers(one_client, label_filter=None):
+    # get hosts (VMs)
+    try:
+        vm_pool = one_client.vmpool.infoextended(-2, -1, -1, 3)
+    except Exception as e:
+        raise AnsibleError("Something happened during XML-RPC call, this was original exception: {e}".format(e=to_native(e)))
+
+    result = []
+
+    # iterate over hosts
+    for vm in vm_pool.VM:
+        server = vm.USER_TEMPLATE
+
+        labels = []
+        if vm.USER_TEMPLATE.get('LABELS'):
+            labels = vm.USER_TEMPLATE.get('LABELS').split(',')
+
+        # filter by label
+        if label_filter is not None:
+            if label_filter not in labels:
+                continue
+
+        server['name'] = vm.NAME
+        server['LABELS'] = labels
+        server['v4_first_ip'] = _get_vm_ipv4(vm)
+        server['v6_first_ip'] = _get_vm_ipv6(vm)
+
+        result.append(server)
+
+    return result
+
+
+class InventoryModule(BaseInventoryPlugin, Constructable):
+    NAME = 'opennebula'
+
+    def verify_file(self, path):
+        valid = False
+        if super(InventoryModule, self).verify_file(path):
+            if path.endswith(('opennebula.yaml', 'opennebula.yml', 'one.yaml', 'one.yml')):
```

Comment:
Is it intentional that `theonlyone.yml` is also matched by this?

Reply:
In such case, we should remove one.yaml and one.yml suffixes

Reply:
You could also replace `one.yaml` by `.one.yaml`. Then it only matches `the.only.one.yml` (which I think is acceptable).

Reply:
Hi, I removed it completely